libtransmission/session.h

```diff
@@ -379,7 +379,7 @@ struct tr_session
     std::string resume_dir;
     std::string torrent_dir;
 
-    std::vector<tr_blocklistFile*> blocklists;
+    std::vector<BlocklistFile*> blocklists;
```

Comment:
Now that BlocklistFile is a C++ class, it would be better to make this a vector of some managed type so that it can clean up after itself & `closeBlocklists()` could be replaced with just a `clear()` call.

Could we make this a vector of either `BlocklistFile` or `std::unique_ptr<BlocklistFile>`? 

The latter is probably simpler to implement.

The former would require we be careful to not accidentally move-construct or move-assign a blocklist file since doing so would probably close the fd_.... actually regardless of what we do with this vector, BlocklistFile should  `= delete` those operations anyway just to protect `fd_`.

Reply:
A quick first look, i'd just make it a vector of BlocklistFiles, no need for indirection via vector of unique_ptrs. Vector code will handle destructions.

Reply:
`BlocklistFile`'s `getRuleCount()` is forced to be non-const due to requirement to also load on demand which is non-const. This contradicts with `tr_blocklistGetRuleCount` taking a `const session*`. Using a smart pointer in a container allows to bypass that, having the container const but the contents not const.